onnxruntime/python/onnxruntime_pybind_ortvalue.cc

```diff
@@ -21,6 +21,42 @@ namespace python {
 
 namespace py = pybind11;
 
+namespace {
+std::unique_ptr<OrtValue> OrtValueFromShapeAndType(const std::vector<int64_t>& shape,
+                                                   MLDataType element_type,
+                                                   const OrtDevice& device) {
+  AllocatorPtr allocator;
+  if (strcmp(GetDeviceName(device), CPU) == 0) {
+    allocator = GetAllocator();
+  } else if (strcmp(GetDeviceName(device), CUDA) == 0) {
+#ifdef USE_CUDA
+    if (!IsCudaDeviceIdValid(logging::LoggingManager::DefaultLogger(), device.Id())) {
+      throw std::runtime_error("The provided device id doesn't match any available GPUs on the machine.");
```

Comment:
Can we mention the device name in the exception messages here and below?

Reply:
Current message indicates that it is a GPU. Probably it is fine since user will not confuse CPU vs GPU. Maybe change to CUDA device is better.

Reply:
This is the original code. 
Similar message is used in many places in this file. I could change the message in a follow-up pull request.

Reply:
This one is up to you